examples/pytorch/question-answering/sparseml_utils.py

```diff
@@ -28,36 +29,74 @@ class SparseMLQATrainer(QuestionAnsweringTrainer):
     :param args, kwargs: arguments passed into parent class
     """
 
-    def __init__(self, recipe, teacher=None, distill_hardness=0.5, distill_temperature=2.0, *args, **kwargs):
+    def __init__(
+        self, model_name_or_path, recipes, teacher=None, distill_hardness=0.5, distill_temperature=2.0, *args, **kwargs
+    ):
         super().__init__(*args, **kwargs)
-        self.recipe = recipe
+        self.model_name_or_path = str(model_name_or_path)
+        self.recipes = [recipe for recipe in recipes if recipe]
         self.teacher = teacher
         self.distill_hardness = distill_hardness
         self.distill_temperature = distill_temperature
         self.criterion = torch.nn.CrossEntropyLoss()
 
-        self.manager = None
+        manager = None
+        modifiers = []
+        for recipe in self.recipes:
+            manager = ScheduledModifierManager.from_yaml(recipe, modifiers)
```

Comment:
is there a good reason you need to pass in an empty list for modifiers just to then set it to manage.modifiers? 
Could it not be manager = ScheduledModifierManager.from_yaml(recipe, None) and also remove line 44?

Reply:
The manager.modifiers hold a list of modifiers accumulated from recipes that were "consumed" so far. We need to add the list back when constructing the manager with the next recipe.